### PR TITLE
Nanites purge healing meds but can't be purged (+ a buff to their healing)

### PR DESCRIPTION
**Pull request by ivanmixo**
## About The Pull Request
- Repaths nanites so they don't get purged by hypervene or ozeo (or water)
- Makes nanites purge chemicals that heal brute or burn (also tramadol because I made nanites give its effects)
- Makes nanites HEAL a small amount of toxin damage (you can still take dylo with them by the way)
## Why It's Good For The Game

The goal of these changes is to make nanites more of a sidegrade to the classic BKTT, with some upsides as well as downsides.

Upsides being 
- You get a little more healing overall 
- You only have to carry blood restoring chemicals instead of pill bottles of BKTT

Downsides being

- You may find it hard to keep blood up if you're constantly taking damage
- You can't inject more on the fly, and if you get defiled (should be the only way to get your nanites purged) you have to get a new injector which will take time to make your nanites work again
## Changelog
:cl:
balance: Nanites have been rebalanced to be a sidegrade to BKTT, check the PR for more details
/:cl:

### CODE REVIEW

File: code/modules/reagents/reagents/medical.dm [hunk 1]
```diff
@@ -1275,57 +1275,71 @@
...
 	description = "These are a batch of construction nanites altered for in-vivo replication. They can heal wounds using the iron present in the bloodstream. Medical care is recommended during injection."
 	color = COLOR_REAGENT_MEDICALNANITES
 	custom_metabolism = 0
 	scannable = TRUE
 	taste_description = "metal, followed by mild burning"
 	overdose_threshold = REAGENTS_OVERDOSE * 1.2 //slight buffer to keep you safe
+	purge_list = list(
+		/datum/reagent/medicine/bicaridine,
+		/datum/reagent/medicine/kelotane,
+		/datum/reagent/medicine/tramadol,
```
Comment (QualityVan) [inline code/modules/reagents/reagents/medical.dm:1289]: Why tram but not oxy?

**Comment by Looking4bros** [comment]
It's funny that it purges all the normal temperature normalization chems, Kelotane and Dermaline. So if you get set on fire you'll have to eat Leporazine or a lot of Kelotane/Dermaline. otherwise you'll take 1-4 damage per tick due to being too hot. Could use drinks, like Iced beer, or a milkshake, or just ice itself, from the soda dispenser. Surt also protects your body temperature from changing from being on fire tiles.

**Comment by timothymtorres** [comment]
You realize the defiler sting is not instant?

**Comment by Tommy-211** [comment]
This is perfect i cant wait to grief 5 million marines with this thank you

**Review by Lumipharon — APPROVED** [review]
Probs still need to address effective infinite blood meme, but a separate pr I guess.

**Comment by Dolth** [comment]
> Upsides being
> 
> You get a little more healing overall
> You only have to carry blood restoring chemicals instead of pill bottles of BKTT
> Downsides being
> 
> You may find it hard to keep blood up if you're constantly taking damage
> You can't inject more on the fly, and if you get defiled (should be the only way to get your nanites purged) you have to get a new injector which will take time to make your nanites work again

Unless I misunderstood something;

Nanites always have been more healing than BKTT, you always had to worry about your blood.
You always needed another inject if you got defiled/purged.

So essentially this is powercreeping and it didn't even go through testmerging, I wonder why I still bother.
I can already tell we're going to see nanites vali that ignore oze and heal for great values with an even easier loadout setup.

**Comment by Fawcksy** [comment]
This is not a sidegrade and to label it as such is misleading. Even moderate combat will easily outpace the blood restoration of both quick-clot and isotonic solution. This could be helped by possibly:
• Making excess nutriment convert into blood after you are full
• Making nanites eat 25-50% less blood
• Making them still usable with T2-3 healing chems (Meralyne, Demaline, Neuraline, RR)

Furthermore, making them unpurgeable could cause problems. If I take nanites, but later decide I don't want them anymore, I'm basically stuck with them forever if they can't be purged.

**Comment by ivanmixo** [comment]
You can go for sleeper dialysis

**Comment by Dolth** [comment]
> This is just not true.
> 
> BKTT heals 2.3 brute or burn per tick (1.8 if you are below 20u of bica or kelo), 0.4 toxin per tick.
> 
> Before nanites healed 2 of each damage per tick, while healing no toxin (it actually gave you toxin when it healed you)
> 
> With this PR nanites heal 3 per damage type, with 0.15 toxin heals per tick (assuming you're above 5u of nanites).
> 
> This might _seem_ like a buff, however you can no longer stack nanites with other healing chemicals, meaning you don't get to almost double your heals by stacking nanites and BKTT (for example; you could stack your heals to much higher heights). You're kinda forced to just commit to nanites and nothing else.

That's absolutely irrelevent to what I said. You heal 3 instead of 4.2, essentially the tradeoff for 1.2 less healing is not being able to be purged and a way easier loadout to manage which is MASSIVE. 

But either way I just answered to your "upsides" and "downsides";
Nanites always have been more healing than BKTT, you always had to worry about your blood.
You always needed another inject if you got defiled/purged.

These are no downsides from this PR since they were already downsides before so essentially, little weaker heal for a huge QOL.

But again, I am using this now and healing a ton of HP per seconds with vali and nanites so sure let's keep it that way I don't mind.

**Pull request merged**